src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -374,7 +375,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     private void unpack(File outputFile) throws NoSuchArchiverException {
         UnArchiver unarchiver = this.archiverManager.getUnArchiver(outputFile);
         unarchiver.setSourceFile(outputFile);
-        unarchiver.setDestDirectory(this.outputDirectory);
+        if (unarchiver instanceof GZipUnArchiver) {
```

Comment:
I think you should remove that condition and always set both destFile and destDirectory. I did look at implementation of the various unarchivers, and it seems like all of them either pick a directory or the file (depending on what they're able to compress), but never both. This will also enable BZip, XZ and more by the way (which currently would have the same issue as GZip).